packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -777,3 +789,33 @@ export function showWarningNotification(type: string, args: any[]) {
   const showMessage = nls.localize(type, ...args);
   vscode.window.showWarningMessage(showMessage);
 }
+
+export const reportExtensionPackStatus = () => {
```

Comment:
🍠 This flow is pretty much exactly how I though It'd work.  I do think these function don't belong in this already freggin' giant file. 

Maybe a new module under `src/telemetry/ExtensionPackStatus.ts`